openpgp/packet/opaque.go

```diff
@@ -76,7 +76,10 @@ func (or *OpaqueReader) Next() (op *OpaquePacket, err error) {
 	op = &OpaquePacket{Tag: uint8(tag), Reason: err}
 	err = op.parse(contents)
 	if err != nil {
-		consumeAll(contents)
+		_, errCon := consumeAll(contents)
+		if errCon != nil {
+			panic(errCon)
+		}
```

Comment:
I don't think you can panic here (since we're reading user data). And since we're already returning an error (since we're inside an `if err != nil`) I wouldn't worry too much that we're ignoring an error here. In fact I would venture to guess that if this errors, it's possibly for that same reason that the error above happened (e.g. maybe we're reading from a broken stream and the data was truncated?)

Reply:
Yes, since the above error comes directly from an `ioutil.ReadAll`, it probably gives the reader error, same as `consumeAll`. I guess I was thinking about the case `err == EOF` which may be wrongly ignored by caller functions (since `ReadAll` returns `err == nil` if successful): Since `OpaqueReader.Next()` does not have documented error behavior in the above comment, caller functions may think `EOF` is fine, even if `consumeAll` failed with something serious. 
In any case, I'll be removing this panic.